cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java

```diff
@@ -173,6 +195,84 @@ protected void addPartition(PartitionKey key, String path, boolean explorable, M
   }
 
   @Override
+  public PartitionConsumerResult consumePartitions(PartitionConsumerState partitionConsumerState) {
+    Set<Long> previousInProgress = partitionConsumerState.getVersionsToCheck();
+    Set<Long> noLongerInProgress = setDiff(previousInProgress, tx.getInProgress());
+
+    List<Iterator<Partition>> partitionIterators = Lists.newArrayList();
+    for (Long txId : noLongerInProgress) {
+      Scanner scanner = partitionsTable.readByIndex(WRITE_PTR_COL, Bytes.toBytes(txId));
+      partitionIterators.add(new PartitionIterator(scanner));
+    }
+
+    long readPointer = tx.getReadPointer();
+    // no read your own writes (partitions)
+    Scanner scanner = partitionsTable.scanByIndex(WRITE_PTR_COL,
+                                                  Bytes.toBytes(partitionConsumerState.getStartVersion()),
+                                                  Bytes.toBytes(readPointer + 1));
```

Comment:
Thanks for making the change. I know that is the current implementation in tephra, but it is nowhere documented. That means it can change with the next version of tephra, but we would not know, because the logic here would simply become non-deterministic with regards to the current transaction, but we would never see an error.